frontend/src/app/pages/exercises/exercise/shared/simulation/signaller-modal/details-modal/signaller-modal-provide-vehicles-editor/signaller-modal-provide-vehicles-editor.component.ts

```diff
@@ -0,0 +1,208 @@
+import {
+    selectCurrentTime,
+    selectTransferPoints,
+    selectVehicleTemplates,
+    selectVehicles,
+} from 'src/app/state/application/selectors/exercise.selectors';
+import { NgbPopover } from '@ng-bootstrap/ng-bootstrap';
+import { MessageService } from 'src/app/core/messages/message.service';
+import { groupBy } from 'lodash-es';
+import { SignallerModalDetailsService } from '../signaller-modal-details.service';
+
+@Component({
+    selector: 'app-signaller-modal-provide-vehicles-editor',
+    templateUrl: './signaller-modal-provide-vehicles-editor.component.html',
+    styleUrls: ['./signaller-modal-provide-vehicles-editor.component.scss'],
+    standalone: false,
+})
+export class SignallerModalProvideVehiclesEditorComponent
+    implements OnInit, OnChanges, OnDestroy
+{
+    @Input() simulatedRegionId!: UUID;
+    @Input() transferBehaviorId!: UUID;
+
+    @ViewChild('selectVehiclePopover')
+    selectVehiclePopover!: NgbPopover;
+    @ViewChild('selectTargetPopover')
+    selectTargetPopover!: NgbPopover;
+
+    public get canSend() {
+        return (
+            this.selectedTarget !== null &&
+            Object.values(this.vehicleAmounts).every((amount) => amount >= 0)
+        );
+    }
+
+    private hotkeyLayer!: HotkeyLayer;
+    selectTargetHotkey = new Hotkey('Z', false, () =>
+        this.selectTargetPopover.open()
+    );
+    submitHotkey = new Hotkey('Enter', false, () => this.startTransfer());
+
+    vehicleTemplates$ = this.store.select(selectVehicleTemplates);
+    vehicleAmounts: ResourceDescription = {};
+
+    availableVehicles$!: Observable<ResourceDescription>;
+
+    availableTargets$!: Observable<SearchableDropdownOption[]>;
+    selectedTarget: SearchableDropdownOption | null = null;
+
+    loading = false;
+
+    constructor(
+        private readonly exerciseService: ExerciseService,
+        private readonly store: Store<AppState>,
+        private readonly detailsModal: SignallerModalDetailsService,
+        private readonly hotkeysService: HotkeysService,
+        private readonly messageService: MessageService
+    ) {}
+
+    ngOnInit() {
+        this.hotkeyLayer = this.hotkeysService.createLayer();
+        this.hotkeyLayer.addHotkey(this.selectTargetHotkey);
+        this.hotkeyLayer.addHotkey(this.submitHotkey);
+    }
+
+    ngOnChanges() {
+        const vehicles$ = this.store.select(selectVehicles);
+        const currentTime$ = this.store.select(selectCurrentTime);
+
+        this.availableVehicles$ = combineLatest([vehicles$, currentTime$]).pipe(
+            map(([vehicles, currentTime]) =>
+                Object.values(vehicles).filter(
+                    (vehicle) =>
+                        isInSpecificSimulatedRegion(
+                            vehicle,
+                            this.simulatedRegionId
+                        ) && isUnoccupiedImmutable(vehicle, currentTime)
```

Comment:
Note to self: Create an issue:

The vehicle the leading personnel belongs to has no occupation and can therefore be sent to other simulated regions which should not happen.